test/drenv/envfile_test.py

```diff
@@ -69,6 +77,46 @@ def test_validate_missing_addons(tmpdir):
         envfile.load(f, addons_root=str(tmpdir))
 
 
+def test_driver():
```

Comment:
Tests using the valid_yaml must create fake addons before loading the file. You can add:

    def test_driver(tmpenv):
        create_addons(tmpdir)

With this the test fail with:

```
________________________________________________________ test_driver _________________________________________________________

    def test_driver():
        f = io.StringIO(valid_yaml)
>       env = envfile.load(f)

drenv/envfile_test.py:82: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
drenv/envfile.py:39: in load
    _validate_env(env, addons_root)
drenv/envfile.py:62: in _validate_env
    _validate_profile(profile, addons_root)
drenv/envfile.py:118: in _validate_profile
    _validate_worker(worker, profile, addons_root, i)
drenv/envfile.py:138: in _validate_worker
    _validate_addon(addon, env, addons_root, args=[env["name"]])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

addon = {'name': 'addon1'}, env = {'addons': [], 'cni': 'auto', 'container_runtime': '', 'cpus': 2, ...}
addons_root = 'addons', args = ['dr1']

    def _validate_addon(addon, env, addons_root, args=()):
        if "name" not in addon:
            raise ValueError(f"Missing addon 'name': {addon}")
    
        addon_dir = os.path.join(addons_root, addon["name"])
        if not os.path.isdir(addon_dir):
>           raise MissingAddon(addon["name"])
E           drenv.envfile.MissingAddon: addon1

drenv/envfile.py:147: MissingAddon
```

Reply:
I posted #985 minimizing the boilerplate needed to crate new envfile tests.